scripts/source-maintenance.sh

```diff
@@ -66,7 +89,11 @@ for FILENAME in `git ls-files`; do
 	#
 	# Code Style formatting maintenance
 	#
-        if test "${ASVER}"; then
+	for SCRIPT in `git ls-files scripts/maintenance/`; do
+		${SCRIPT} "${FILENAME}" >"${FILENAME}.new" &&
+			updateIfChanged "${FILENAME}" "${FILENAME}.new" "by ${SCRIPT}"
```

Comment:
As you know, I would like us to avoid piling modifications on top of failures:

```suggestion
			updateIfChanged "${FILENAME}" "${FILENAME}.new" "by ${SCRIPT}" || return
```

If we cannot agree on that, I suggest making error ignorance configurable (with a conservative default), similar to `make -k`. I can contribute the code necessary to make that happen.

Reply:
Accepting that the rm set of changes is fatal to the entire maintenance process now, the "failures" that remain are just minor nits where skipping the script is acceptable.

There should *not* be any dependency between these scripts. Specifically because we cannot predetermine which order they will be run. ie. If script A depends on script B; then they should be combined into one, or A calls B and only A placed in the scripts/maintenance/ directory for auto-execution.

Reply:
> the "failures" that remain are just minor nits where skipping the script is acceptable

I disagree with that assertion. Applying modifications after any plugin failure, especially after a failure that leaves partial output _may_ make triaging the situation a lot more difficult. It all depends on the specifics of the failure and on whether the failure was even noticed. We cannot reliably predict those things.

I can easily imagine cases where continuing is harmful _and_ cases where continuing is desirable. Moreover, I am willing to contribute proper support for both. Is not resistance to that solution counter-productive?

> There should not be any dependency between these scripts.

Yes, we agree on plugin independence -- no need to remind me of that principle in this context. The principle is a good one (at least for now), but it does not address my concern.

Reply:
If you wish to put the work in yourself feel free to do so. I do not think this issue is serious enough to spend any more time on.

Please note that "return" is silent so if there is an actual *problem* happening something else is needed.

Reply:
> If you wish to put the work in yourself feel free to do so.

Done at f50627e.